File: Java-Core/src/main/java/prompto/literal/DictEntry.java
package prompto.literal;

import prompto.expression.IExpression;
import prompto.utils.CodeWriter;


public class DictEntry {
	
	IExpression key;
	IExpression value;
	
	
	public DictEntry(IExpression key, IExpression value) {
		this.key = key;
		this.value = value;
	}

	public IExpression getKey() {
		return key;
	}
	
	public IExpression getValue() {
		return value;
	}
	
	@Override
	public String toString() {
		return key.toString() + ':' + value.toString();
	}

	public void toDialect(CodeWriter writer) {
		key.toDialect(writer);
		writer.append(':');
		value.toDialect(writer);
	}
	
}

File: Java-CodeServer/src/main/resources/js/prompto/literal/DictEntry.js
function DictEntry(key, value) {
	this.key = key;
	this.value = value;
	return this;
}

DictEntry.prototype.toString = function() {
	return this.key.toString() + ':' + this.value.toString();
};

DictEntry.prototype.toDialect = function(writer) {
    this.key.toDialect(writer);
    writer.append(':');
    this.value.toDialect(writer);
};

exports.DictEntry = DictEntry;
